### DISCUSSION TITLE
How could I free the memory of  struct lysp_xxx after compilation ?

### DISCUSSION THREAD

**Discussion by lyzliyuzhi91** [CESNET/libyang#2089] [Q&A]
Once after I compile explicitly, I will not access the lysp_xxx struct, eg lysp_module/lysp_node， any more.
And I use libyang in a IOT device, which there is shortage for memory resource.  So I want to free the unused infomation after compilation.
Thanks

**Comment by michalvasko** [comment]
During the initial libyang v2 design we envision the ability to free all parsed schema modules and keep only the compiled ones. However, we never got to actually implementing it. The reason not being just that there were tasks with higher priority but also because it turned out that the parsed module is needed for more tasks than we expected (the first that comes to my mind is the tree printer).

So, I could (once I have some spare time) try to add support for this and checks to all the unsupported tasks to handle the situation gracefully but the unanswered question remains that I am not sure how usable libyang would be in such a state, whether some basic features would not be disabled.
